commonknowledge/mapped
# feat: adds runworker command that will restart the worker on code changes

## Description
Adds the command, and updates the run configuration to use this instead of the underlying `procrastinate worker` command.

## Motivation and Context
When developing Mapped, it's very annoying to have to remember to restart the worker when changing background job code.

## How Can It Be Tested?
1. Start the run configuration "Django background worker" (or "Run Mapped!")
2. Wait for the worker to start up (`Launching a worker on all queues`)
3. Toggle the log level of `procrastinate.worker` in `local_intelligence_hub/settings.py` from `INFO` to `DEBUG` (or vice versa)
4. Observe that the worker restarts. If the log level is `DEBUG`, there should be `procrastinate.worker.wait_for_job:172 # Waiting for new jobs on all queues` log statements. If the log level is `INFO`, there should not be.

You can also check that breakpoints work in the job functions, e.g. `deferred_import_many`.
